### DISCUSSION TITLE
Hello how to i get starting balance ?

### DISCUSSION THREAD

**Discussion by gprethesh** [nguyenphuminh/JeChain#72] [General]
i have created wallet address for my localhost. how do i get tokens or balance 

but when i sent http://localhost:3000/get_balance via postman.

```
{
  "params": {
    "address": "bc77654bd2d7449885b6041f5ae596535c857d0dde90f149856a9f9493897bfc"
  }
}

```

i get 

```
{
    "success": false,
    "payload": null,
    "error": {
        "message": "Invalid option."
    }
}


```

now how do i call genesis block or something  ?

```
const EC = require("elliptic").ec,
  ec = new EC("secp256k1");
const crypto = require("crypto"),
  SHA256 = (message) =>
    crypto.createHash("sha256").update(message).digest("hex");

const Block = require("./block");
const Transaction = require("./transaction");
const { FIRST_ACCOUNT } = require("../config.json");

function generateGenesisBlock() {
  return new Block(1, Date.now(), [], 1, "", FIRST_ACCOUNT);
}

module.exports = generateGenesisBlock;

```

FIRST_ACCOUNT is never mentioned.

**Comment by gprethesh** [comment]
how do i send token from 1 account to another ?

```
const axios = require("axios");
const EC = require("elliptic").ec;
const ec = new EC("secp256k1");

const privateKey =
  "09cca4a40afd67a8aa1454e0ed355b1bde161ca4a6d6349200923209519230f1";
const keyPair = ec.keyFromPrivate(privateKey);

const Transaction = require("../src/core/transaction");
const rpc = require("./rpc/rpc");


const recipientAddress =
  "3346291f44b1c948d43c8a7bed30428cf4ac464b029bef26f0d300e6b33fad2c";
const amountToSend = 100;

const myContract = {};

const transaction = new Transaction(recipientAddress, amountToSend, 100000, {
  scBody: myContract,
});

Transaction.sign(transaction, keyPair);

sendTransaction(transaction);

axios
  .post("http://localhost:3000/sendTransaction", {
    params: {
      transaction: transaction,
    },
  })
  .then((response) => {
    console.log(`Response`, response.data);
  })
  .catch((error) => {
    console.error(`2ndError`, error.response.data);
  });
```

where do i import `sendTransaction(transaction);` ?


i don't know how to require sendTransaction so i used axios but i'm getting 

` {
  success: false,
  payload: null,
  error: { message: 'Invalid request.' }
}`

**Comment by nguyenphuminh** [comment]
Sorry for the late reply, I did not realize that there is a new discussion in here.

> but when i sent http://localhost:3000/get_balance via postman.

You get "invalid option" error when the method you are calling does not exist. In this case I think you are sending it as a GET request rather than POST.

> now how do i call genesis block or something ?

you don't call it, if you are starting a new chain, the node will generate a new genesis block by calling that for you. Or you can request blocks from other nodes you are connected to by enabling "ENABLE_CHAIN_REQUEST" (that's if you have other nodes to connect to).

> how do i send token from 1 account to another ?

Here's an example for creating a transaction:
```js
// Import the transaction class and EC somewhere here

const transaction = new Transaction("paste receiver address here", "amount to send here", "gas fee", {}, nonce_as_number);
// Oof, I forgot to add an RPC to get nonce, will release a patch soon

Transaction.sign(transaction, ec.keyFromPrivate("Your private key goes here"));
// Alternatively you can use the `signTransaction` rpc method to sign a transaction.

// Send the transaction
axios
    .post("http://localhost:3000/sendTransaction", {
        params: {
            transaction: Transaction.serialize(transaction),
        },
    })
    .then((response) => {
        console.log(`Response`, response.data);
    })
    .catch((error) => {
        console.error(`2ndError`, error.response.data);
    });
```

Don't use the `sendTransaction` function, the `http://localhost:3000/sendTransaction` API already used it behind the scenes.

Things are kind of complicated currently because I haven't put too much notice on the UX and libs to interact with JeChain, too many things to add to the infra :sweat_smile: If you have any question, feel free to ask! Will be more alerted about discussions from this point on.

**Comment by gprethesh** [comment]
cool, thanks for responding..
